src/main/java/com/demo/classroom/Entity/Course.java

```diff
@@ -0,0 +1,41 @@
+package com.demo.classroom.Entity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.OneToMany;
+import jakarta.validation.constraints.NotBlank;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Course {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(updatable = false, nullable = false)
+    private Long id;
+
+    @NotBlank(message = "Course must have a title.")
+    @Column(nullable = false)
+    private String title;
+
+    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<Book> books = new ArrayList<>();
+
+    @ManyToMany(mappedBy = "studentCourses")
+    private List<Student> students = new ArrayList<>();
+
+    @ManyToMany(mappedBy = "teacherCourses")
+    private List<Teacher> teachers = new ArrayList<>();
```

Comment:
let's rename to `courses` to maintain consistency.
```suggestion
    @ManyToMany(mappedBy = "courses")
    private List<Student> students = new ArrayList<>();

    @ManyToMany(mappedBy = "courses")
    private List<Teacher> teachers = new ArrayList<>();
```

Reply:
done